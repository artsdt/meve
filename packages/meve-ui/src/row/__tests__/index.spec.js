import Row from '..'
import Col from '../../col'
import { mount } from '@vue/test-utils'

test('test col span', () => {
  const wrapper = mount({
    components: {
      [Row.name]: Row,
      [Col.name]: Col,
    },
    template: `
      <m-row>
        <m-col :span="8">span: 8</m-col>
        <m-col :span="8">span: 8</m-col>
        <m-col :span="8">span: 8</m-col>
      </m-row>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row gutter', () => {
  const wrapper = mount({
    components: {
      [Row.name]: Row,
      [Col.name]: Col,
    },
    template: `
      <m-row :gutter="10">
        <m-col :span="8">span: 8</m-col>
        <m-col :span="8">span: 8</m-col>
        <m-col :span="8">span: 8</m-col>
      </m-row>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row offset', () => {
  const wrapper = mount({
    components: {
      [Row.name]: Row,
      [Col.name]: Col,
    },
    template: `
      <m-row>
        <m-col :span="6">span: 6</m-col>
        <m-col :offset="2" :span="8">offset: 2 span: 8</m-col>
        <m-col :span="8">span: 8</m-col>
      </m-row>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row responsive', () => {
  const wrapper = mount({
    components: {
      [Row.name]: Row,
      [Col.name]: Col,
    },
    template: `
      <m-row>
        <m-col :xs="24" :sm="12" :md="8" :lg="4" :xl="2">Responsive</m-col>
      </m-row>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row responsive pass object', () => {
  const wrapper = mount({
    components: {
      [Row.name]: Row,
      [Col.name]: Col,
    },
    template: `
      <m-row>
        <m-col
          :xs="{ span: 22, offset: 2 }"
          :sm="{ span: 10, offset: 2 }"
          :md="{ span: 6, offset: 2 }"
          :lg="{ span: 3, offset: 1 }"
          :xl="{ span: 2 }"
        >
          Responsive
        </m-col>
      </m-row>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row click event', async () => {
  const wrapper = mount(Row)
  await wrapper.trigger('click')

  expect(wrapper.emitted().click.length).toBe(1)
})

test('test col click event', async () => {
  const wrapper = mount(Col)
  await wrapper.trigger('click')

  expect(wrapper.emitted().click.length).toBe(1)
})
