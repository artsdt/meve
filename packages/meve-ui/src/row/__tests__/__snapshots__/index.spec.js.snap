// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test col span 1`] = `
"<div class=\\"m-row m--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px 0px;\\">
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 0px;\\">span: 8</div>
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 0px;\\">span: 8</div>
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 0px;\\">span: 8</div>
</div>"
`;

exports[`test row gutter 1`] = `
"<div class=\\"m-row m--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px -10px;\\">
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 5px;\\">span: 8</div>
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 5px;\\">span: 8</div>
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 5px;\\">span: 8</div>
</div>"
`;

exports[`test row offset 1`] = `
"<div class=\\"m-row m--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px 0px;\\">
  <div class=\\"m-col m--box m-col--span-6\\" style=\\"padding: 0px 0px;\\">span: 6</div>
  <div class=\\"m-col m--box m-col--span-8 m-col--offset-2\\" style=\\"padding: 0px 0px;\\">offset: 2 span: 8</div>
  <div class=\\"m-col m--box m-col--span-8\\" style=\\"padding: 0px 0px;\\">span: 8</div>
</div>"
`;

exports[`test row responsive 1`] = `
"<div class=\\"m-row m--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px 0px;\\">
  <div class=\\"m-col m--box m-col--span-24 m-col--span-xs-24 m-col--span-sm-12 m-col--span-md-8 m-col--span-lg-4 m-col--span-xl-2\\" style=\\"padding: 0px 0px;\\">Responsive</div>
</div>"
`;

exports[`test row responsive pass object 1`] = `
"<div class=\\"m-row m--box\\" style=\\"justify-content: flex-start; align-items: flex-start; margin: 0px 0px;\\">
  <div class=\\"m-col m--box m-col--span-24 m-col--span-xs-22 m-col--offset-xs-2 m-col--span-sm-10 m-col--offset-sm-2 m-col--span-md-6 m-col--offset-md-2 m-col--span-lg-3 m-col--offset-lg-1 m-col--span-xl-2\\" style=\\"padding: 0px 0px;\\">
    Responsive
  </div>
</div>"
`;
